components/raftstore-v2/src/router/message.rs

```diff
@@ -208,65 +94,92 @@ impl StoreTick {
     }
 }
 
-/// Raft command is the command that is expected to be proposed by the
-/// leader of the target raft group.
-#[derive(Debug)]
-pub struct RaftCommand<S: Snapshot> {
+/// Command that can be handled by raftstore.
+pub struct RaftCommand {
     pub send_time: Instant,
     pub request: RaftCmdRequest,
-    pub ch: ResponseChannel<S>,
 }
 
-impl<S: Snapshot> RaftCommand<S> {
-    #[inline]
-    pub fn new(request: RaftCmdRequest, ch: ResponseChannel<S>) -> RaftCommand<S> {
+impl RaftCommand {
+    pub fn new(request: RaftCmdRequest) -> Self {
         RaftCommand {
+            send_time: Instant::now(),
             request,
+        }
+    }
+}
+
+pub struct ReadCommand {
+    pub cmd: RaftCommand,
+    pub ch: ReadChannel,
+}
+
+impl ReadCommand {
+    #[inline]
+    pub fn new(request: RaftCmdRequest, ch: ReadChannel) -> Self {
+        Self {
+            cmd: RaftCommand::new(request),
+            ch,
+        }
+    }
+}
+
+pub struct WriteCommand {
+    pub cmd: RaftCommand,
+    pub ch: WriteChannel,
```

Comment:
Prefer to name it `WriteResponseChannel` or some name to represent it used for sending responses, not requests.